From ae66baa23455df9f7593b98c5c2f02818dbaf41b Mon Sep 17 00:00:00 2001
From: James Hughes <JamesH65@users.noreply.github.com>
Date: Wed, 11 Sep 2019 14:57:18 +0100
Subject: [PATCH] drm/vc4: Fix for margins in composite/SDTV mode
 (#3223)

Margins were incorrectly assumed to be setup in SDTV mode, but were
not actually done, so this make the setup non-conditional on mode.

Signed-off-by: James Hughes <james.hughes@raspberrypi.org>
---
 drivers/gpu/drm/vc4/vc4_firmware_kms.c | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

--- a/drivers/gpu/drm/vc4/vc4_firmware_kms.c
+++ b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
@@ -1612,14 +1612,9 @@ vc4_fkms_connector_init(struct drm_devic
 		connector->interlace_allowed = 0;
 	}
 
-	/* Create and attach TV margin props to this connector.
-	 * Already done for SDTV outputs.
-	 */
-	if (fkms_connector->display_type != DRM_MODE_ENCODER_TVDAC) {
-		ret = drm_mode_create_tv_margin_properties(dev);
-		if (ret)
-			goto fail;
-	}
+	ret = drm_mode_create_tv_margin_properties(dev);
+	if (ret)
+		goto fail;
 
 	drm_connector_attach_tv_margin_properties(connector);
 
